### PR TITLE
Mark the create embed steps in embedding hub as done from embed flow

### PR DESCRIPTION
**Pull request by heypoom**
Closes EMB-743

Marks the "Create an embed" step as completed in the Embedding Hub when users interact with the Get Code functionality in the embed flow. It also updates the UI to align with the proposed design from Alessio e.g. locked steps.

## How to verify

- Go to Embedding Hub
- If JWT is disabled or unconfigured, "Embed in production" step will be locked.
  - "Configure SSO authentication to unlock this step." hint text should show, and "Create an embed" button should be disabled.
- Clicking on "create an embed" on the "create a test embed" step should take you to embed flow with "use existing user session" auth method selected.
- Clicking on "create an embed" on the "embed in production" step should take you to embed flow with "SSO" auth method selected.
- Clicking on "Copy Code" or "Done" in the embed flow will mark each step as completed.

## Changes

- Automatically mark embedding hub steps as done when users copy code or click "Done" in the Get Code step
- Mark step as complete based on the auth methods:
  - useExistingUserSession → marks test embed step complete
  - SSO → marks production embed step complete
- Add locked states and info alerts for embedding hub steps
- Update alert styling with proper flex layout and text wrapping
- Add test_tube and check_filled icons
- Show appropriate CTAs and disable buttons when steps are locked
- Add new backend settings for tracking embedding hub step completion

### CODE REVIEW

**Comment by heypoom** [comment]
This change is part of the following stack:

- #62759 ◀

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

File: enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetup.tsx [hunk 1]
```diff
@@ -17,27 +19,32 @@ import { SdkIframeEmbedSetupProvider } from "./SdkIframeEmbedSetupProvider";
...
   const acceptSimpleEmbedTerms = () =>
     updateSettings({ "show-simple-embed-terms": false });
 
+  function handleEmbedDone() {
+    // Embedding Hub: track step completion
+    const settingKey: SettingKey = settings.useExistingUserSession
+      ? "embedding-hub-test-embed-snippet-created"
+      : "embedding-hub-production-embed-snippet-created";
+
+    updateSettings({ [settingKey]: true });
```
Comment (heypoom) [inline enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetup.tsx:43]: I previously extracted this, but reverted it as updating the settings in place is more greppable.

File: enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetupProvider.tsx [hunk 2]
```diff
@@ -34,6 +35,7 @@ interface SdkIframeEmbedSetupProviderProps {
 export const SdkIframeEmbedSetupProvider = ({
   children,
 }: SdkIframeEmbedSetupProviderProps) => {
+  const location = useLocation();
```
Comment (heypoom) [inline enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetupProvider.tsx:38]: Makes unit testing easier as we can mock useLocation.

File: enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup.unit.spec.tsx [hunk 3]
```diff
@@ -1,39 +1,8 @@
 import userEvent from "@testing-library/user-event";
 
-import {
```
Comment (heypoom) [inline enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup.unit.spec.tsx:3]: Move this to `./test-setup.tsx` so the main test file isn't super long.

File: enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup.unit.spec.tsx [hunk 4]
```diff
@@ -175,34 +144,3 @@ describe("Embed flow > usage terms card", () => {
     expect(matchingRequest).toBeDefined();
   });
 });
-
-async function waitForPutRequests() {
```
Comment (heypoom) [inline enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup.unit.spec.tsx:179]: Also moved this to test-setup.tsx so it can be shared.

File: enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/test-setup.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+  setupUpdateSettingsEndpoint();
+  setupUpdateSettingEndpoint();
+
+  renderWithProviders(<SdkIframeEmbedSetup />, {
+    storeInitialState: createMockState({
+      settings: createMockSettingsState({
+        "show-simple-embed-terms": options?.showSimpleEmbedTerms ?? true,
+        "enable-embedding-simple": options?.simpleEmbeddingEnabled ?? false,
+        "jwt-enabled": options?.jwtReady ?? false,
+        "jwt-configured": options?.jwtReady ?? false,
```
Comment (heypoom) [inline enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/test-setup.tsx:33]: This whole file is extracted from the above test file and it's the same as before, the only change is this jwt-enabled and jwt-configured setting that's needed for the SSO test.

File: frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHub.tsx [hunk 6]
```diff
@@ -25,6 +26,11 @@ export const EmbeddingHub = () => {
   // eslint-disable-next-line no-unconditional-metabase-links-render -- This link only shows for admins.
   const embedJsDocsUrl = useDocsUrl("embedding/embedded-analytics-js");
 
+  const lockedSteps: Partial<Record<EmbeddingHubStepId, boolean>> = {
+    // Production embeds requires JWT to be configured.
+    "embed-production": !completedSteps["secure-embeds"],
```
Comment (heypoom) [inline frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHub.tsx:48]: We decided to not lock the row column security step, so it's more clear that the step is optional.

File: frontend/src/metabase/embedding/embedding-hub/utils/embedding-hub-steps.ts [hunk 7]
```diff
@@ -84,6 +84,10 @@ export const getEmbeddingHubSteps = (): EmbeddingHubStep[] => {
       srcSet: "app/assets/img/embedding_hub_row_column_security@2x.png 2x",
       alt: t`Screenshot of configuring row and column security`,
     },
+    infoAlert: {
+      type: "always",
+      message: t`Log in using SSO to create a user with attributes before setting up \n row and column security.`,
```
Comment (heypoom) [inline frontend/src/metabase/embedding/embedding-hub/utils/embedding-hub-steps.ts]: The `\n` makes the message breaks more natural.

image

File: enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetupProvider.tsx [hunk 8]
```diff
@@ -137,11 +144,26 @@ export const SdkIframeEmbedSetupProvider = ({
...
-      setRawSettings({ ...settings, ...persistedSettings });
+      setRawSettings({
+        ...settings,
+        ...persistedSettings,
+
+        // Override the persisted settings if `auth_method` is specified.
+        // This is used for Embedding Hub.
+        ...(authMethodOverride !== null && {
+          useExistingUserSession: authMethodOverride === "user_session",
+        }),
```
Comment (heypoom) [inline enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/SdkIframeEmbedSetupProvider.tsx:155]: If a user has selected `useExistingUserSession` before but navigates to the embedding hub's embed in production step, we should force using SSO.

File: frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHubChecklist/EmbeddingHubChecklist.module.css [hunk 9]
```diff
@@ -7,6 +7,16 @@
   ) !important;
 }
 
+/* Override styles for locked steps */
+.lockedControl:not([data-css-specificity-hack="🪗"]) {
+  background: linear-gradient(
+    to right,
+    var(--mb-color-bg-light) 3rem,
```
Comment (heypoom) [inline frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHubChecklist/EmbeddingHubChecklist.module.css:14]: Locked steps should have different color of accordion control.

image

File: frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHubChecklist/EmbeddingHubChecklist.tsx [hunk 10]
```diff
@@ -111,3 +145,21 @@ const accordionClassNames = {
...
+const InfoAlert = ({ message, icon }: { message: string; icon?: IconName }) => (
+  <Alert
+    variant="info"
+    px="md"
+    py="sm"
+    bg="var(--mb-color-bg-light)"
+    bd="1px solid var(--mb-color-border)"
+  >
+    <Flex gap="sm" align="flex-start">
+      {icon && <Icon name={icon} c="var(--mb-color-text-secondary)" mt={4} />}
```
Comment (heypoom) [inline frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHubChecklist/EmbeddingHubChecklist.tsx:158]: Aligns the icon for multi-line inputs. Also works for single line ones.

image

File: frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHub.tsx [hunk 11]
```diff
@@ -42,6 +43,11 @@ export const EmbeddingHub = () => {
     },
   });
 
+  const lockedSteps: Partial<Record<EmbeddingHubStepId, boolean>> = {
+    // Production embeds requires JWT to be configured.
```
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHub.tsx:47]: no, that is obvious and you can see that by looking at the code. the fact that production embeds requires JWT to be configured is more non-obvious.

File: enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup-embedding-hub.unit.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,52 @@
+import userEvent from "@testing-library/user-event";
+
+import { screen } from "__support__/ui";
+
+import { setup, waitForUpdateSetting } from "./test-setup";
+
+const mockLocation = (search: string) =>
+  jest.mock("react-use", () => ({ useLocation: jest.fn(() => ({ search })) }));
```
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup-embedding-hub.unit.spec.tsx:8]: we need to mock it dynamically and re-define it across test cases.

File: frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHubChecklist/EmbeddingHubStepActions.tsx [hunk 13]
```diff
@@ -49,7 +52,7 @@ export const EmbeddingHubStepActions = ({
         if (action.to) {
           return (
             <Link key={index} to={action.to}>
-              <Button variant={action.variant ?? "outline"}>
+              <Button variant={action.variant ?? "outline"} disabled={isLocked}>
```
[the remark this answers is not in the document]
Comment (heypoom) [reply frontend/src/metabase/embedding/embedding-hub/components/EmbeddingHubChecklist/EmbeddingHubStepActions.tsx:55]: Verified that I can't click on "Create an embed" when disabled is set.

File: enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup-embedding-hub.unit.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+describe("Embed flow > embedding hub step completion tracking", () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
```
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup-embedding-hub.unit.spec.tsx]: fixed in https://github.com/metabase/metabase/pull/62759/commits/3befb49a1fb28de44689ce08ccbd1382f0b59f5c

File: enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup-embedding-hub.unit.spec.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (heypoom) [reply enterprise/frontend/src/metabase-enterprise/embedding_iframe_sdk_setup/components/tests/SdkIframeEmbedSetup-embedding-hub.unit.spec.tsx:8]: you can keep recommending that, that's fine. this dynamic mocking is more of an edge case.

File: frontend/src/metabase/embedding/embedding-hub/utils/embedding-hub-steps.ts (hunk 7 above)
Comment (npretto) [reply frontend/src/metabase/embedding/embedding-hub/utils/embedding-hub-steps.ts]: I don't think we should hardcode new lines in translations strings. That would be weird for translators and they may remove it. If design wants it to go on a new line I think our best bet will be putting a max-width on the inner container, but I doubt this is worth it as it may do worse on other languages

Comment (heypoom) [reply frontend/src/metabase/embedding/embedding-hub/utils/embedding-hub-steps.ts]: @npretto Good point, I didn't think about that. What about if we just add a `<br>` and add a context `{0} is a line break`?

Comment (npretto) [reply frontend/src/metabase/embedding/embedding-hub/utils/embedding-hub-steps.ts]: I'm still not a fan of randomly putting a new line in the string, what's the purpose? if it's "this should go on a new line after xxx pixels" then it should be a max-width, otherwise how will the translators know where to put it?
Even with the max-width we should check for text direction

Comment (heypoom) [reply frontend/src/metabase/embedding/embedding-hub/utils/embedding-hub-steps.ts]: Right, the content width will be completely different by languages and `max-width` may only look good on English. Maybe we just remove the newline altogether 😞

Comment (heypoom) [reply frontend/src/metabase/embedding/embedding-hub/utils/embedding-hub-steps.ts]: removed in https://github.com/metabase/metabase/pull/62759/commits/652ffaea9948b9b583302553042503bad3aa5538

**Review by npretto — APPROVED** [review]

**Comment by npretto** [comment]
@heypoom i approved this so we can merge it, we can change the settings name to be embed-js specific later so we don't keep this open 🙏

**Comment by heypoom** [comment]
@npretto Thank you!
